tensorflow/compiler/tf2tensorrt/convert/ops/variable_ops.cc

```diff
@@ -30,20 +30,124 @@ namespace tensorflow {
 namespace tensorrt {
 namespace convert {
 
+struct VarAttributes {
+  TensorShapeProto shape_proto;
+  TensorShape shape;
+  string name;
+  DataType dtype;
+  string shared_name;
+  string container;
+};
+
+template <typename T, bool is_resource>
+Status ReadVariableHelper(const OpConverterParams* params,
+                          const VarAttributes& attrs,
+                          TRT_ShapedWeights* weights) {
+  Tensor tensor(attrs.dtype, attrs.shape);
+  auto ctx = params->converter->context();
+  TRT_ENSURE(ctx != nullptr);
+  auto tensor_flat = tensor.flat<T>();
+
+  // Clone function library runtime in order to get a mutable library
+  // definition to add and run a function with the variable operation.
+  auto lib = ctx->function_library();
+  std::unique_ptr<FunctionLibraryDefinition> lib_def;
+  std::unique_ptr<ProcessFunctionLibraryRuntime> lib_pflr;
+  FunctionLibraryRuntime* lib_clone;  // Not owned.
+  TF_RETURN_IF_ERROR(lib->Clone(&lib_def, &lib_pflr, &lib_clone));
+
+  // Create function definition.
+  FunctionDef fdef;
+  std::vector<Tensor> args;
+  string func_name = attrs.name + "/func";
+  if (is_resource) {
+    // Create input tensor with the resource handle.
+    const auto& inputs = params->inputs;
+    const TRT_TensorOrWeights& handle = inputs.at(0);
+    args.emplace_back(handle.resource());
+
+    fdef = FunctionDefHelper::Define(
+        func_name,                                             // Name
+        {"in: resource"},                                      // Args
+        {absl::StrCat("out: ", DataTypeString(attrs.dtype))},  // Returns
+        {},                                                    // Attr def
+        // Nodes
+        {{{attrs.name},
+          "ReadVariableOp",
+          {"in"},  // Name of the Placeholder or VarHandleOp
+          {{"dtype", attrs.dtype}}},
+         {{"out"}, "Identity", {attrs.name}, {{"T", attrs.dtype}}}});
+  } else {
+    fdef = FunctionDefHelper::Define(
+        func_name,                                             // Name
+        {},                                                    // Args
+        {absl::StrCat("out: ", DataTypeString(attrs.dtype))},  // Returns
+        {},                                                    // Attr def
+        // Nodes
+        {{{attrs.name},
+          "VariableV2",
+          {},
+          {{"dtype", attrs.dtype},
+           {"shape", attrs.shape_proto},
+           {"container", attrs.container},
+           {"shared_name", attrs.shared_name}}},
+         {{"out"}, "Identity", {attrs.name}, {{"T", attrs.dtype}}}});
+  }
+
+  // Add function definition to the library.
+  lib_def->AddFunctionDef(fdef);
```

Comment:
error: ignoring return value of function declared with 'warn_unused_result' attribute [-Werror,-Wunused-result]